### PR TITLE
[FIX] pos_loyalty: enable free product for Next Order Coupon program

### PR DESCRIPTION
**Pull request by pedrambiria**
Before this commit, the free product reward feature in a Next Order Coupon program was not functioning as expected. This commit fixes this issue, ensuring that free product rewards are correctly applied in Next Order Coupon programs.

opw-3703485

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/pos_loyalty/static/src/tours/PosLoyaltyTour.js [hunk 1]
```diff
@@ -279,3 +279,41 @@ ProductScreen.check.totalAmountIs('1.00');
...
 
 Tour.register('PosLoyaltyTour10', { test: true, url: '/pos/web' }, getSteps());
+
+startSteps();
+
+ProductScreen.do.clickHomeCategory();
+ProductScreen.do.confirmOpeningPopup();
+
+ProductScreen.do.clickPartnerButton();
+ProductScreen.do.clickCustomer('AAA Partner');
```
Comment (caburj) [inline addons/pos_loyalty/static/src/tours/PosLoyaltyTour.js:289]:
```suggestion
ProductScreen.do.clickCustomer('AAA Partner');
PosLoyalty.check.customerIs('AAA Partner');
```
wait for the customer to be reflected in the UI before clicking the product.

File: addons/pos_loyalty/static/src/tours/PosLoyaltyTour.js [hunk 2]
```diff
@@ -279,3 +279,41 @@ ProductScreen.check.totalAmountIs('1.00');
...
+ProductScreen.do.clickDisplayedProduct('Product Test');
+ProductScreen.do.pressNumpad('3')
```
Comment (caburj) [inline addons/pos_loyalty/static/src/tours/PosLoyaltyTour.js:291]:
```suggestion
ProductScreen.exec.addOrderline('Product Test', '3');
```
Use addOrderline. It waits for the orderline to be rendered properly (with correct quantity), before going to the next step.

**Review by caburj — CHANGES_REQUESTED** [review]
Apply the suggestions also on the 2nd tour.

**Comment by pedrambiria** [comment]
@caburj Thanks for your comments. I applied them.

**Comment by caburj** [comment]
robodoo r+

**Pull request closed without merging**